### PR TITLE
Add a check task to plugins to enforce up-to-date version numbers

### PR DESCRIPTION
**Pull request by SamCarlberg**
This will force future updates to plugins to bump the version number

### CODE REVIEW

**Comment by SamCarlberg** [comment]
@JLLeitschuh Any thoughts on why travis wouldn't be able to run `git` commands? Might be a path thing?

**Comment by JLLeitschuh** [comment]
Not sure, but in general, wouldn't this whole thing be easier with resource files?

**Comment by SamCarlberg** [comment]
How would resource files be helpful? We still need to make sure the version numbers are bumped regardless of what files they're stored in

**Comment by AustinShalit** [comment]
I agree with Jonathan.  He is saying that instead of parsing a java file, we should parse a resource file.  That way it is less likely to parse incorrectly.

**Comment by JLLeitschuh** [comment]
Resource files can be modified/rewritten using the gradle task that already exists.
This is the example you want:
http://mrhaki.blogspot.com/2010/11/gradle-goodness-add-filtering-to.html

File: plugins/plugins.gradle.kts [hunk 1]
```diff
@@ -3,6 +3,10 @@ import org.gradle.jvm.tasks.Jar
 
 subprojects {
     afterEvaluate {
+        /** The name of the plugin. This is, by convention, entirely lower-case. */
+        val pluginName: String = project.name
+        /** The plugin name as a proper noun, with the first character capitalized */
+        val proper = Character.toUpperCase(pluginName[0]) + pluginName.substring(1)
```
Comment (AustinShalit) [inline plugins/plugins.gradle.kts]: Can we do something smarter here so it follows the name of the plugin rather than blindly capitalizing the first char?

Comment (SamCarlberg) [reply plugins/plugins.gradle.kts]: That would be ideal. Project names are always lowercase eg `networktables`, `powerup`, `cameraserver` and cannot be reassigned. I suppose a property might be a good enough solution, even though it's not very enforceable

File: plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/BasePluginVersion.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  /**
+   * The version of the Base plugin.
+   */
+  public static final String VERSION = "1.5.0";
+
+  private BasePluginVersion() {
+    throw new UnsupportedOperationException();
+  }
+}
```
Comment (AustinShalit) [inline plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/BasePluginVersion.java]: EOL @ EOF

File: plugins/cameraserver/src/main/java/edu/wpi/first/shuffleboard/plugin/cameraserver/CameraServerPluginVersion.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+  /**
+   * The version of the CameraServer plugin.
+   */
+  public static final String VERSION = "1.5.0";
+
+  private CameraServerPluginVersion() {
+    throw new UnsupportedOperationException();
+  }
+}
```
Comment (AustinShalit) [inline plugins/cameraserver/src/main/java/edu/wpi/first/shuffleboard/plugin/cameraserver/CameraServerPluginVersion.java]: EOL @ EOF

File: plugins/networktables/src/main/java/edu/wpi/first/shuffleboard/plugin/networktables/NetworkTablesPluginVersion.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+  /**
+   * The version of the NetworkTables plugin.
+   */
+  public static final String VERSION = "1.5.0";
+
+  private NetworkTablesPluginVersion() {
+    throw new UnsupportedOperationException();
+  }
+}
```
Comment (AustinShalit) [inline plugins/networktables/src/main/java/edu/wpi/first/shuffleboard/plugin/networktables/NetworkTablesPluginVersion.java]: EOL @ EOF

File: plugins/powerup/src/main/java/edu/wpi/first/shuffleboard/plugin/powerup/PowerUpPluginVersion.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+  /**
+   * The version of the PowerUp plugin.
+   */
+  public static final String VERSION = "1.5.0";
+
+  private PowerUpPluginVersion() {
+    throw new UnsupportedOperationException();
+  }
+}
```
Comment (AustinShalit) [inline plugins/powerup/src/main/java/edu/wpi/first/shuffleboard/plugin/powerup/PowerUpPluginVersion.java]: EOL @ EOF

File: plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/BasePluginVersion.java (hunk 2 above)
Comment (SamCarlberg) [reply plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/BasePluginVersion.java]: Irrelevant for an autogenerated file IMO

Comment (AustinShalit) [reply plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/BasePluginVersion.java]: We should edit line six to say that it is autogenerated.

File: plugins/powerup/src/main/java/edu/wpi/first/shuffleboard/plugin/powerup/PowerUpPluginVersion.java (hunk 5 above)
Comment (JLLeitschuh) [reply plugins/powerup/src/main/java/edu/wpi/first/shuffleboard/plugin/powerup/PowerUpPluginVersion.java]: You know spotless can enforce this right?

**Comment by SamCarlberg** [comment]
Going to close this. It's too fragile. I'll add a PR template to prompt the author to check that they've updated the plugin versions

**Pull request closed without merging**